Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/viewmodel/SubmissionViewModel.kt

```diff
@@ -48,11 +49,26 @@ class SubmissionViewModel : ViewModel() {
         SubmissionRepository.testResultReceivedDate
     val deviceUiState: LiveData<DeviceUIState> =
         SubmissionRepository.deviceUIState
+    val visitedCountries: LiveData<List<String>> =
+        SubmissionRepository.visitedCountries
+    val consentToFederation: LiveData<Boolean> =
+        SubmissionRepository.consentToFederation
+
+    private fun <T> notNullGuard(value: T?): T {
+        if (value == null) {
+            throw RequiredParameterNotSetException()
+        }
+        return value
+    }
 
     fun submitDiagnosisKeys(keys: List<TemporaryExposureKey>) = viewModelScope.launch {
         try {
             _submissionState.value = ApiRequestState.STARTED
-            SubmissionService.asyncSubmitExposureKeys(keys)
+            SubmissionService.asyncSubmitExposureKeys(
+                notNullGuard(visitedCountries.value),
+                notNullGuard(consentToFederation.value),
```

Comment:
As this feature is not complete I assume that proper handling for the RequiredParameterNotSetException follows in a separate PR

Reply:
exacly, this is just placeholder code untill the ui components are built